Makefile

```diff
@@ -14,12 +21,28 @@ lint:
 	golangci-lint run ./...
 
 test:
-	TEST_FREELIST_TYPE=hashmap go test -timeout 30m -v -coverprofile cover.out -covermode atomic
-	TEST_FREELIST_TYPE=hashmap go test -v ./cmd/bbolt
+	@echo "hashmap freelist test"
+	TEST_FREELIST_TYPE=hashmap go test -v ${TESTFLAGS} -timeout 30m
+	TEST_FREELIST_TYPE=hashmap go test -v ${TESTFLAGS} ./cmd/bbolt
+
+	@echo "array freelist test"
+	TEST_FREELIST_TYPE=array go test -v ${TESTFLAGS} -timeout 30m
+	TEST_FREELIST_TYPE=array go test -v ${TESTFLAGS} ./cmd/bbolt
+
+test-simulate:
+	@echo "hashmap freelist test"
+	TEST_FREELIST_TYPE=hashmap go test -v ${TESTFLAGS} -test.run="TestSimulate_(100op|1000op)"
 
 	@echo "array freelist test"
+	TEST_FREELIST_TYPE=array go test -v ${TESTFLAGS} -test.run="TestSimulate_(100op|1000op)"
 
-	@TEST_FREELIST_TYPE=array go test -timeout 30m -v -coverprofile cover.out -covermode atomic
-	@TEST_FREELIST_TYPE=array go test -v ./cmd/bbolt
+coverage:
+	@echo "hashmap freelist test"
+	TEST_FREELIST_TYPE=hashmap go test -v -timeout 30m \
+		-coverprofile cover-freelist-hashmap.out -covermode atomic
+
+	@echo "array freelist test"
+	TEST_FREELIST_TYPE=array go test -v -timeout 30m \
+		-coverprofile cover-freelist-array.out -covermode atomic
 
-.PHONY: race fmt test lint
+.PHONY: fmt test test-simulate lint
```

Comment:
Does it add value to have both here: "test" and "test-simulate" ? 
It seems to me that test-simulate is subset of the 'test'. 

If we cannot afford running all tests in the --race mode, probably --short and 'test-simulate' in the race mode is offering a cross-cutting coverage.

Reply:
Agreed. 

Since we still need to support `array-freelist` although I agree to deprecate it, I would suggest to only run test with flag `--short` for `array-freelist`. 

FYI. 
- https://github.com/etcd-io/etcd/blob/e73f55d4e94666c99558baa2fd4e365aeaca4dc4/scripts/test.sh#L100
- https://github.com/etcd-io/etcd/blob/e73f55d4e94666c99558baa2fd4e365aeaca4dc4/tests/e2e/etcd_config_test.go#L33
- https://github.com/etcd-io/bbolt/blob/b654ce922133ff49bb385297f30835ca357bac5f/simulation_test.go#L34-L36

Reply:
I deleted the `test-simulate` since it is duplicate.
And I split the `make test` into `make test-freelist-hashmap` and `make test-freelist-array` so that we can use `EXTRA_TESTFLAGS=-short` for `array`.